### PR TITLE
Improvements for `dshackle` chart

### PR DESCRIPTION
**Pull request by cristidas**
- add definition missing for `dshackle.httpPort`
- adding `annotations` support for configuring a Service of type `LoadBalancer`
- adding support for setting up the external and internal listening ports - for SSL support on 443

### CODE REVIEW

File: charts/dshackle/templates/_helpers.tpl [hunk 1]
```diff
@@ -60,3 +60,10 @@ Create the name of the service account to use
 {{- default "default" .Values.serviceAccount.name }}
 {{- end }}
 {{- end }}
+
+{{/*
+HTTP port
```
Comment (eskp) [inline charts/dshackle/templates/_helpers.tpl:65]: HTTPS?

**Review by eskp — CHANGES_REQUESTED** [review]
The default values for `externalHttpPort` and `internalHttpPort` should be set in `values.yaml`

File: charts/dshackle/templates/_helpers.tpl (hunk 1 above)
Comment (cristidas) [reply charts/dshackle/templates/_helpers.tpl:65]: We could also change to HTTPS, but throughout this chart `http` is used to label that specific traffic

**Review by eskp — APPROVED** [review]

File: charts/dshackle/templates/_helpers.tpl (hunk 1 above)
Comment (OleksandrUA) [reply charts/dshackle/templates/_helpers.tpl:65]: `http` here represents internal port which is not HTTPS, internally they communicate over HTTP on port 8545

File: charts/dshackle/templates/service.yaml [hunk 2]
```diff
@@ -4,11 +4,13 @@ metadata:
...
     {{- include "dshackle.labels" . | nindent 4 }}
+  annotations:
+    {{ toYaml .Values.service.annotations | nindent 4 }}
 spec:
   type: {{ .Values.service.type }}
   ports:
-    - port: {{ .Values.httpPort }}
-      targetPort: http
+    - port: {{ .Values.externalHttpPort }}
+      targetPort: {{ .Values.internalHttpPort }}
```
Comment (OleksandrUA) [inline charts/dshackle/templates/service.yaml:13]: @cristidas I think if we plan to open PR to back-merge it we should make as less changes as possible. I would suggest to introduce `externalHttpPort` but keep using `httpPort` as internal one

**Review by sanbotto — APPROVED** [review]

File: charts/dshackle/templates/service.yaml (hunk 2 above)
Comment (cristidas) [reply charts/dshackle/templates/service.yaml:13]: @OleksandrUA it wouldn't be all these changes in a PR, and I've already opened one only for the changes related to `service.type: LoadBalancer`. We'll see after that if we'll need to refactor the changes for the ports and create a PR to upstream

**Pull request merged**